### DISCUSSION TITLE
Compatability with newer versions of Gym?

### DISCUSSION THREAD

**Discussion by Bladerunner18IS** [aidudezzz/deepbots#140] [Q&A] [answered]
For a project I am working on with deepbots, I need to use the new 'Sequence' space from the Gym framework, which was not available in Gym 0.21.

Is it possible to run deepbots with an updated version of gym? Or perhaps is there an alternative to the sequence space, which is used because the amount of data going into the observation_space is variable.

[https://gymnasium.farama.org/api/spaces/composite/#sequence](url)

Any help would be appreciated, thanks.

**Comment by tsampazk** [comment, marked as answer]
Hello there! I can't think of any particular reason why you can't use the space you need. Deepbots doesn't force you to use any particular space, for example in some of our examples we don't use gym spaces at all but simple integers. It mainly depends on the RL agent implementation you are using and whether that is compatible with the space. 

Deepbots requires gym to be installed so that the produced environment is recognized as a gym environment by the most popular RL frameworks. So, if your agent is compatible with the particular space you want to use, you can go ahead and import gymnasium and define the space normally.

Let me know if that works for you :)
